verilog/formatting/formatter_test.cc

```diff
@@ -7441,6 +7441,120 @@ static constexpr FormatterTestCase kFormatterTestCases[] = {
      "  generic_type_name_t third;\n"
      "`endif\n"
      "} type_t;\n"},
+    // Continuation comment alignment
+    {"`define BAR 1 // A\n"
+     "module foo(); // B\n"
+     "wire baz;     // C\n"
+     "endmodule:foo // D\n",
+     "`define BAR 1 // A\n"
+     "module foo ();  // B\n"
+     "  wire baz;  // C\n"
+     "endmodule : foo  // D\n"},
+    {"`define BAR 1 // A\n"
+     "module foo(); // B\n"
+     "              // B.1\n"
+     "              // B.2\n"
+     "wire baz;     // C\n"
+     "              // C.1\n"
+     "              // C.2\n"
+     "endmodule:foo // D\n"
+     "              // D.1\n"
+     "              // D.2\n",
+     "`define BAR 1 // A\n"
+     "module foo ();  // B\n"
+     "                // B.1\n"
+     "                // B.2\n"
+     "  wire baz;  // C\n"
+     "             // C.1\n"
+     "             // C.2\n"
+     "endmodule : foo  // D\n"
+     "                 // D.1\n"
+     "                 // D.2\n"},
+    {"// W\n"
+     "`define BAR 1 // A\n"
+     "   // X\n"
+     "module foo(); // B\n"
+     "              // B.1\n"
+     "              // B.2\n"
+     " // Y\n"
+     "wire baz;     // C\n"
+     "              // C.1\n"
+     "              // C.2\n"
+     "    // Z\n"
+     "endmodule:foo // D\n"
+     "              // D.1\n"
+     "              // D.2\n",
+     "// W\n"
+     "`define BAR 1 // A\n"
+     "// X\n"
+     "module foo ();  // B\n"
+     "                // B.1\n"
+     "                // B.2\n"
+     "  // Y\n"
+     "  wire baz;  // C\n"
+     "             // C.1\n"
+     "             // C.2\n"
+     "  // Z\n"
+     "endmodule : foo  // D\n"
+     "                 // D.1\n"
+     "                 // D.2\n"},
+    {"module foo( // A\n"
+     "            // A.1\n"
+     "// X\n"
+     "input wire i1 [a:b], // B\n"
+     "                     // B.1\n"
+     "input [c:d] i2, // C\n"
+     "                // C.1\n"
+     "\n"
+     "// Y\n"
+     "output reg o1 // D\n"
+     "              // D.1\n"
+     ");endmodule:foo\n",
+     "module foo (  // A\n"
+     "              // A.1\n"
+     "    // X\n"
+     "    input wire       i1[a:b],  // B\n"
+     "                               // B.1\n"
+     "    input      [c:d] i2,       // C\n"
+     "                               // C.1\n"
+     "\n"
+     "    // Y\n"
+     "    output reg o1  // D\n"
+     "                   // D.1\n"
+     ");\n"
+     "endmodule : foo\n"},
+
+    {"               // comment1\n"
+     "module foo();  // A\n"
+     "                // comment2\n"
+     "              // comment3\n"
+     "wire baz;     // B\n"
+     "             // comment4\n"
+     "               // comment5\n"
+     "              // comment6\n"
+     "endmodule:foo // C\n"
+     "               // comment7\n"
+     "              // comment8\n",
+     "// comment1\n"
+     "module foo ();  // A\n"
+     "  // comment2\n"
```

Comment:
The difference is that starting columns of comments in comment-only lines are different than the starting column of a comment in a line above them. All comments in this test case are independent (i.e. none are "continuation comments"). 
My intent was to just not break manually aligned comments, so I tried to not introduce too much heuristics.

I've modified the code to allow +/- 1 column difference, added a new test case, and comments with explanations.

As for user intent detection, it could be a good idea to check which column is closer: first column (i.e. indent) of the current scope (→ not continuation), or starting column of a comment in the line above (→ continuation).

Reply:
I like the idea of refining the column alignment depending on current vs previous line. For now, I'd just add this as a comment to potentially revisit later.

Reply:
Comment added